File: Stack & Queue/leetcode_71.py
# Problem Link ====>>https://leetcode.com/problems/simplify-path/description/

class Solution:
    def simplify_path(self, path: str) -> str:

        stack = []
        res = ""

        for char in path:
            if char == '/':
                continue

            temp = ""
            while char != '/' and path.index(char) < len(path):
                temp += char
                char = path[path.index(char) + 1]

            if temp == ".":
                continue
            elif temp == "..":
                if stack:
                    stack.pop()
            else:
                stack.append(temp)

        while stack:
            res = "/" + stack.pop() + res

        return "/" if not res else res

if __name__ == "__main__":
    input_path = input("Enter a path to simplify: ")
    solution = Solution()
    simplified_path = solution.simplify_path(input_path)
    print("Simplified path:", simplified_path)

File: Stack & Queue/leetcode_71.cpp
// Problem Link ====>>https://leetcode.com/problems/simplify-path/description/

#include<bits/stdc++.h>
using namespace std;

class Solution
{
public:
    string simplifyPath(string path)
    {
        stack<string> st;
        string res;

        for (int i = 0; i < path.size(); i++){
            if(path[i] == '/'){
                continue;
            }

            string temp;
            while(i<path.size() && path[i] !='/'){
                temp += path[i];
                ++i;
            }

            if(temp=="."){
                continue;
            }
            else if(temp==".."){
                if(!st.empty()){
                    st.pop();
                }
            }
            else{
                st.push(temp);
            }
        }

        while (!st.empty())
        {
            res = "/" + st.top() + res;
            st.pop();
        }

        if (res.size() == 0)
        {
            return "/";
        }

        return res;
    }
};

int main(){
    string inputPath;
    cout << "Enter a path to simplify: ";
    getline(cin, inputPath);

    Solution solution;

    string simplifiedPath = solution.simplifyPath(inputPath);

    cout << "Simplified path: " << simplifiedPath << endl;

    return 0;
}